### PR TITLE
remove traversal second attempt

### PR DESCRIPTION
**Pull request by bbrehm**
Second attempt to remove traversal, this time with smaller diff (because we retain head/headOption/last/lastOption). Tests pass locally with appropriate locally published upstream.

### CODE REVIEW

**Comment by Liyw979** [comment]
Hi, Why do you remove traversals here?

File: dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/SourcesToStartingPoints.scala [hunk 1]
```diff
@@ -43,7 +42,7 @@ class SourceTravsToStartingPointsTask[NodeType](sourceTravs: Traversal[NodeType]
     val sources: List[StoredNode] = sourceTravs
       .flatMap(_.toList)
       .collect { case n: StoredNode => n }
-      .dedup
+      .distinct
```
Comment (fabsx00) [inline dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/SourcesToStartingPoints.scala:45]: Looks like we're breaking all queries containing `dedup` here. Wondering if we could just introduce an alias.

File: joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/LambdaTests.scala [hunk 2]
```diff
@@ -126,14 +119,14 @@ class LambdaTests extends JavaSrcCode2CpgFixture {
           val fallbackLocal = cpg.method.name(".*lambda.*").local.name("fallback").head
           fallbackClosureBinding.closureBindingId shouldBe fallbackLocal.closureBindingId
 
-          fallbackClosureBinding.outE.collectAll[Ref].map(_.inNode()).l match {
+          fallbackClosureBinding.outE.asScala.collectAll[Ref].map(_.inNode()).l match {
```
Comment (fabsx00) [inline joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/LambdaTests.scala:122]: Can we introduce `collectAll` on `outE` and friends?

**Comment by fabsx00** [comment]
@bbrehm are you on this?

File: joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/LambdaTests.scala (hunk 2 above)
Comment (mpollmeier) [reply joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/LambdaTests.scala]: How about an extension method `collectAll` on `java.util.Iterator`? I guess that's what fabs had in mind here.

**Pull request merged**